src/Controller/ControllerFactory.php

```diff
@@ -268,7 +268,7 @@ protected function coerceStringToType(string $argument, ReflectionNamedType $typ
             case 'float':
                 return is_numeric($argument) ? (float)$argument : null;
             case 'int':
-                return ctype_digit($argument) ? (int)$argument : null;
+                return ctype_digit($argument) || filter_var($argument, FILTER_VALIDATE_INT) ? (int)$argument : null;
```

Comment:
The filter_var() check doesn't cover the ctype_digit() check too?

Reply:
filter_var doesn't validate leading 0's unless you enable octal, but that will re-interpret the value.

Reply:
Unless we have existing tests for casting strings with leading zeros to int I don't think we need to support it.

Reply:
We do, The first test checks `02` which is why filter_var() fails on its own.

Reply:
😢

Reply:
Yeah, I'm not sure we want to try and support octals long term. But its there right now :shrug:

Reply:
We don't support octals. It's just casting the string to int which strips the leading 0s.

Reply:
The leading 0 handling could be interpreted as octal support though. I'm ok breaking this, it feels like a scenario that isn't going to be common in userland code.

Reply:
I'm ok with it as well. The unit tests were added to support the code we used to validate.

Reply:
Let's remove the `ctype_digit()` check and the test with leading zeros then.